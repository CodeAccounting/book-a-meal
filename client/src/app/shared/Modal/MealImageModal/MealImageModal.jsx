import React from 'react';
import PropTypes from 'prop-types';
import { RenderFileInput } from '../../FormComponents';

/**
 * @exports
 * @function MealImageModal
 * @param {object} props
 * @returns {component} AddMealImageModal
 */
const MealImageModal = (props) => {
  const {
    formerImgURL, mealId, editMeal, updating
  } = props;

  return (
    <RenderFileInput
      id="imageURL"
      name="imageURL"
      mealId={mealId}
      label="Meal Image"
      updating={updating}
      formerImgURL={formerImgURL}
      successCallBack={imageURL => editMeal(mealId, { imageURL }, true)}
    />
  );
};

MealImageModal.propTypes = {
  formerImgURL: PropTypes.string.isRequired,
  updating: PropTypes.bool.isRequired,
  mealId: PropTypes.string.isRequired,
  editMeal: PropTypes.func.isRequired
};

export default MealImageModal;
