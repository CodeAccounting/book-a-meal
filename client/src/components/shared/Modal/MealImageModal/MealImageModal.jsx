import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { RenderFileInput } from '../../FormComponents';

/**
 * @exports
 * @class MealImageModal
 * @extends Component
 * @returns {JSX} MealImageModal
 */
class MealImageModal extends Component {
  /**
   * @memberof MealImageModal
   * @param {string} imageURL
   * @returns {JSX} MealImageModal
   */
  updateMealImage = imageURL => this.props.editMeal(this.props.mealId, { imageURL }, true);

  /**
   * @memberof MealImageModal
   * @returns {JSX} MealImageModal
   */
  render() {
    const { formerImgURL, mealId, updating } = this.props;

    return (
      <RenderFileInput
        id="imageURL"
        name="imageURL"
        mealId={mealId}
        label="Meal Image"
        updating={updating}
        formerImgURL={formerImgURL}
        successCallBack={this.updateMealImage}
      />
    );
  }
}

MealImageModal.propTypes = {
  formerImgURL: PropTypes.string.isRequired,
  updating: PropTypes.bool.isRequired,
  mealId: PropTypes.string.isRequired,
  editMeal: PropTypes.func.isRequired
};

export default MealImageModal;
