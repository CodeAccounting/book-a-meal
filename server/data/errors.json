{
  "401": "Unauthorized",
  "403": "Forbidden",
  "404": "Not Found",
  "422": "Validation Error",
  "500": "Internal Server Error",
  "empty": "Empty PUT Requests Are Not Allowed"
}