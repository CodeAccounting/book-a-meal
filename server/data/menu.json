[
  {
    "userId": "8356954a-9a42-4616-8079-887a73455a7f",
     "menuId": "a9fa6cb3-9f5e-46fa-b641-388f898ca824",
     "date": "2018-05-06",
     "meals": [
        "81211c24-51c0-46ec-b1e0-18db55880958",
        "36d525d1-efc9-4b75-9999-3e3d8dc64ce3",
        "baa0412a-d167-4d2b-b1d8-404cb8f02631",
        "f9eb7652-125a-4bcb-ad81-02f84901cdc3"
     ],
     "created": "2018-05-06T14:34:90.000",
     "updated": "2018-05-06T14:34:90.000"
  },
  {
    "userId": "8356954a-9a42-4616-8079-887a73455a7f",
     "menuId": "1adfcfe7-c66d-42d2-82fd-39c1decd290a",
     "date": "2018-04-06",
     "meals": [
        "a3c35e8f-da7a-4113-aa01-a9c0fc088539",
        "3ec802c6-0a32-4d29-b27b-42e0f4b532dd"
     ],
     "created": "2018-04-06T14:34:90.000",
     "updated": "2018-04-06T14:34:90.000"
  },
  {
    "userId": "ac1b253c-6b33-439b-ab6f-805a4fdd2e05",
     "menuId": "3a4fb034-6e52-4bde-9abc-91119a39478d",
     "date": "2018-05-07",
     "meals": [
        "72a3417e-45c8-4559-8b74-8b5a61be8614"
     ],
     "created": "2018-05-06T14:34:90.000",
     "updated": "2018-05-06T14:34:90.000"
  },
  {
    "userId": "ac1b253c-6b33-439b-ab6f-805a4fdd2e05",
     "menuId": "f43f3d49-a6c9-476d-b65a-1af772ff0f36",
     "date": "2018-04-06",
     "meals": [
        "72a3417e-45c8-4559-8b74-8b5a61be8614",
        "46ced7aa-eed5-4462-b2c0-153f31589bdd"
     ],
     "created": "2018-05-06T14:34:90.000",
     "updated": "2018-05-06T14:34:90.000"
  },
  {
     "userId": "8356954a-9a42-4616-8079-887a73455a7f",
     "menuId": "15421f7a-0f82-4802-b215-e0e8efb6bfb3",
     "date": "2018-04-27",
     "meals": [
        "91b6e41c-0972-4ac5-86da-4ac1f5226e83",
        "f9eb7652-125a-4bcb-ad81-02f84901cdc3",
        "8a65538d-f862-420e-bcdc-80743df06578"
     ],
     "created": "2018-05-06T14:34:90.000",
     "updated": "2018-05-06T14:34:90.000"
  }
]