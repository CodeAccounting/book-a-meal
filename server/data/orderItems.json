[
  {
     "id": 1,
     "mealId": "46ced7aa-eed5-4462-b2c0-153f31589bdd",
     "orderId": "fb097bde-5959-45ff-8e21-51184fa60c25",
     "quantity": 2,
     "createdAt": "2018-04-06T14:34:90.000",
     "updatedAt": "2018-04-06T14:34:90.000"
  },
  {
     "id": 2,
     "mealId": "36d525d1-efc9-4b75-9999-3e3d8dc64ce3",
     "orderId": "fb097bde-5959-45ff-8e21-51184fa60c25",
     "quantity": 1,
     "createdAt": "2018-04-06T14:34:90.000",
     "updatedAt": "2018-04-06T14:34:90.000"
  }, 
  {
    "id": 3,
    "mealId": "72a3417e-45c8-4559-8b74-8b5a61be8614",
    "orderId": "fb097bde-5959-45ff-8e21-51184fa60c25",
    "quantity": 2,
    "createdAt": "2018-04-06T14:34:90.000",
    "updatedAt": "2018-04-06T14:34:90.000"
  },
  {
    "id": 4,
    "mealId": "a3c35e8f-da7a-4113-aa01-a9c0fc088539",
    "orderId": "6ed0963f-9663-4fe2-8ad4-2f06c6294482",
    "quantity": 2,
    "createdAt": "2018-05-06T14:34:90.000",
    "updatedAt": "2018-05-06T14:34:90.000"
  },
  {
    "id": 5,
    "mealId": "baa0412a-d167-4d2b-b1d8-404cb8f02631",
    "orderId": "6ed0963f-9663-4fe2-8ad4-2f06c6294482",
    "quantity": 2,
    "createdAt": "2018-05-06T14:34:90.000",
    "updatedAt": "2018-05-06T14:34:90.000"
  }
]