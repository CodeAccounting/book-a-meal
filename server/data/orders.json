[
  {
     "orderId": "fb097bde-5959-45ff-8e21-51184fa60c25",
     "created": "2018-04-06T14:34:90.000",
     "updated": "2018-04-06T14:34:90.000",
     "mealId": "46ced7aa-eed5-4462-b2c0-153f31589bdd",
     "menuId": "f43f3d49-a6c9-476d-b65a-1af772ff0f36",
     "userId": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
     "deliveryAddress": "4, Church Street, Yaba",
     "deliveryPhoneNo": "+2348134567890",
     "quantity": 2
  },
  {
     "orderId": "ce228787-f939-40a0-bfd3-6607ca8d2e53",
     "created": "2018-05-06T14:34:90.000",
     "updated": "2018-05-06T14:34:90.000",
     "mealId": "36d525d1-efc9-4b75-9999-3e3d8dc64ce3",
     "menuId": "a9fa6cb3-9f5e-46fa-b641-388f898ca824",
     "userId": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
     "deliveryAddress": "4, Church Street, Yaba",
     "deliveryPhoneNo": "+2348134567890",
     "quantity": 2
  },
  {
     "orderId": "e544248c-145c-4145-b165-239658857637",
     "created": "2018-05-07T14:34:90.000",
     "updated": "2018-05-07T14:34:90.000",
     "mealId": "72a3417e-45c8-4559-8b74-8b5a61be8614",
     "menuId": "3a4fb034-6e52-4bde-9abc-91119a39478d",
     "userId": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
     "deliveryAddress": "4, Church Street, Yaba",
     "deliveryPhoneNo": "+2348134567890",
     "quantity": 2
  },
  {
     "orderId": "e5508b87-7975-493d-a900-3d47a69dad03",
     "created": "2018-04-06T14:34:90.000",
     "updated": "2018-04-06T14:34:90.000",
     "mealId": "a3c35e8f-da7a-4113-aa01-a9c0fc088539",
     "menuId": "1adfcfe7-c66d-42d2-82fd-39c1decd290a",
     "userId": "e42c7ce0-958e-4368-adcb-e5d97dbbe5b7",
     "deliveryAddress": "4, Church Street, Yaba",
     "deliveryPhoneNo": "+2348134567890",
     "quantity": 2
  },
  {
     "orderId": "6ed0963f-9663-4fe2-8ad4-2f06c6294482",
     "created": "2018-05-06T14:34:90.000",
     "updated": "2018-05-06T14:34:90.000",
     "mealId": "baa0412a-d167-4d2b-b1d8-404cb8f02631",
     "menuId": "a9fa6cb3-9f5e-46fa-b641-388f898ca824",
     "userId": "e42c7ce0-958e-4368-adcb-e5d97dbbe5b7",
     "deliveryAddress": "4, Church Street, Yaba",
     "deliveryPhoneNo": "+2348134567890",
     "quantity": 2
  }
]