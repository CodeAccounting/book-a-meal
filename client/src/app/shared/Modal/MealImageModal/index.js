import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import MealImageModal from './MealImageModal';
import { editMeal } from '../../../../store/operations/meals';
import { getLastMealItem } from '../../../../store/selectors/meals';

const mapStateToProps = state => ({
  updating: state.meals.working,
  mealId: getLastMealItem(state.meals.items).id,
  formerImgURL: getLastMealItem(state.meals.items).imageURL
});

const mapDispatchToProps = dispatch => bindActionCreators({
  editMeal
}, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(MealImageModal);
