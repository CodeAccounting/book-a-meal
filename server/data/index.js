import users from './users.json';
import meals from './meals.json';
import menu from './menu.json';
import orders from './orders.json';
import errors from './errors.json';

export default {
  users, meals, menu, orders, errors
};
