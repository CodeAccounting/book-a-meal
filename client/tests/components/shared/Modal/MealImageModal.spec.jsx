import React from 'react';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import { initialValues, caterersMealsObj } from '../../../setup/data';
import MealImageModal from '../../../../src/app/shared/Modal/MealImageModal/MealImageModal';
import ConnectedMealImageModal from '../../../../src/app/shared/Modal/MealImageModal';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);
const store = mockStore({
  ...initialValues,
  meals: { ...initialValues.meals, items: caterersMealsObj.meals }
});
const props = {
  formerImgURL: 'image.jpg',
  updating: false,
  mealId: '1234',
  editMeal: jest.fn()
};

describe('MealImageModal', () => {
  afterAll(() => {
    jest.clearAllMocks();
  });

  it('renders correctly', () => {
    const shallowWrapper = shallow(<MealImageModal {...props} />);

    expect(toJson(shallowWrapper)).toMatchSnapshot();
  });

  // it('calls editMeal function on file change', () => {
  //   const editMealMock = jest.fn();
  //   const event = {
  //     target: {
  //       files: [new Blob([''], {
  //         type: 'image/jpeg',
  //         size: '500'
  //       })]
  //     }
  //   };
  //   const comp = (
  //     <Provider store={store}>
  //       <MealImageModal
  //         {...props}
  //         formerImgURL="https://firebasestorage.googleapis.com/v0/b/book-a-meal.appspot.com/o/images%2Fplaceholder-image.jpg?alt=media&token=e688dcde-0496-4a10-a456-0825e5202c62"
  //         editMeal={editMealMock}
  //       />
  //     </Provider>
  //   );
  //   const wrapper = mount(comp);

  //   // const handleUploadSuccessSpy =
  // jest.spyOn(wrapper.find('MealImageModal').instance(), 'handleUploadSuccess');

  //   mocksdk.storage().ref = () => ({
  //     put: () => ({
  //       ref: {
  //         getDownloadURL: () => new Promise(resolve => resolve('http.img.test'))
  //       }
  //     })
  //   });

  //   wrapper.find('input').simulate('change', event);

  //   expect(editMealMock).toHaveBeenCalled();
  // });

  it('renders connected component correctly', () => {
    const comp = (<Provider store={store}><ConnectedMealImageModal {...props} /></Provider>);
    const mountedWrapper = mount(comp);

    expect(toJson(mountedWrapper)).toMatchSnapshot();
    mountedWrapper.unmount();
  });
});
