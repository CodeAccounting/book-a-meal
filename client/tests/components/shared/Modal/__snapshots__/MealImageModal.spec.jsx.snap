// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MealImageModal renders connected component correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(MealImageModal)
    editMeal={[MockFunction]}
    formerImgURL="image.jpg"
    mealId="1234"
    updating={false}
  >
    <MealImageModal
      editMeal={[Function]}
      formerImgURL="https://images.pexels.com/photos/461198/pexels-photo-461198.jpeg?auto=compress&cs=tinysrgb&h=350"
      mealId="36d525d1-efc9-4b75-9999-3e3d8dc64ce3"
      updating={false}
    >
      <Connect(RenderFileInput)
        formerImgURL="https://images.pexels.com/photos/461198/pexels-photo-461198.jpeg?auto=compress&cs=tinysrgb&h=350"
        id="imageURL"
        label="Meal Image"
        mealId="36d525d1-efc9-4b75-9999-3e3d8dc64ce3"
        name="imageURL"
        successCallBack={[Function]}
        updating={false}
      >
        <RenderFileInput
          clearUploadError={[Function]}
          formerImgURL="https://images.pexels.com/photos/461198/pexels-photo-461198.jpeg?auto=compress&cs=tinysrgb&h=350"
          id="imageURL"
          label="Meal Image"
          mealId="36d525d1-efc9-4b75-9999-3e3d8dc64ce3"
          name="imageURL"
          successCallBack={[Function]}
          updating={false}
          uploadError={null}
          uploadImage={[Function]}
          uploading={false}
        >
          <div
            className="form-input form-img-input"
          >
            <label
              htmlFor="imageURL"
            >
              Meal Image
            </label>
            <div
              id="form-image-preview"
            >
              <img
                alt="meal"
                id="meal-form-image"
                src="https://images.pexels.com/photos/461198/pexels-photo-461198.jpeg?auto=compress&cs=tinysrgb&h=350"
              />
            </div>
            <input
              accept="image/*"
              id="imageURL"
              name="imageURL"
              onChange={[Function]}
              type="file"
            />
          </div>
        </RenderFileInput>
      </Connect(RenderFileInput)>
    </MealImageModal>
  </Connect(MealImageModal)>
</Provider>
`;

exports[`MealImageModal renders correctly 1`] = `
<Connect(RenderFileInput)
  formerImgURL="image.jpg"
  id="imageURL"
  label="Meal Image"
  mealId="1234"
  name="imageURL"
  successCallBack={[Function]}
  updating={false}
/>
`;
